### PR TITLE
chore: homogeneous tests setup and close servers.

### PR DESCRIPTION
**Pull request by ldez**
- fix missing `server.Close`
- refactor method to setup tests

### CODE REVIEW

File: providers/dns/servercow/internal/client_test.go [hunk 1]
```diff
@@ -14,7 +14,7 @@ import (
 
 func setupAPIMock() (*Client, *http.ServeMux, func()) {
 	handler := http.NewServeMux()
-	svr := httptest.NewServer(handler)
+	svr := httptest.NewServer(handler) // FIME t.Cleanup(server.Close)
```
Comment (dmke) [inline providers/dns/servercow/internal/client_test.go:17]: Is this FIXME intentional?

**Review by dmke — CHANGES_REQUESTED** [review]
Thank you for bringing `t.Cleanup` to my attention. I didn't know it even existed - this is a great feature (I'll start using this in my own code)!

In `servercow/internal/client_test.go` there's a FIXME left over (I think). Will the tests fail if there's no `t.Cleanup(server.Close)`? The code doesn't look like it would, though.

File: providers/dns/servercow/internal/client_test.go (hunk 1 above)
Comment (ldez) [reply providers/dns/servercow/internal/client_test.go]: oops!

Comment (ldez) [reply providers/dns/servercow/internal/client_test.go]: if you like `t.Cleanup`, you can take a look to `t.Setenv()`

**Review by dmke — APPROVED** [review]

File: providers/dns/servercow/internal/client_test.go (hunk 1 above)
Comment (dmke) [reply providers/dns/servercow/internal/client_test.go]: Neat!

**Pull request merged**